libr/include/r_types.h

```diff
@@ -654,4 +655,18 @@ static inline void r_run_call10(void *fcn, void *arg1, void *arg2, void *arg3, v
 # endif
 #endif
 
+// reference counter
+typedef void (*RFreeCallback)(void *a);
+typedef volatile int RRef;
+
+#define R_REF_NAME refcount
+#define r_ref(x) x->R_REF_NAME++;
+#define r_ref_init(x) x->R_REF_NAME = 1
+#define r_unref(x,f) { assert (x->R_REF_NAME> 0); if (!--x->R_REF_NAME) { f(x); } }
+
+#define R_REF_TYPE RRef refcount;
+#define R_REF_FUNCTIONS(s, n, f) \
```

Comment:
since we are already assuming `r_##n##_ref`/`r_##n##_unref` we may as well assume `r_##n##_free` and avoid passing the additional `f` argument.

Reply:
right

Reply:
i think everything should be freed with just a 'free'.

Reply:
+1